sympy/printing/fcode.py

```diff
@@ -106,6 +106,15 @@ def _get_statement(self, codestring):
 
     def _get_comment(self, text):
         return "! {0}".format(text)
+    
+    #issue 12267
+    def _print_sign(self,func):
+        if func.args[0].is_integer:
+            return "merge(0, isign(1, {0}), {0} == 0)".format(self._print(func.args[0]))
+        elif func.args[0].is_complex:
+            return "merge(cmplx(0d0, 0d0), {0}/abs({0}), abs({0}) == 0d0)".format(self._print(func.args[0]))
```

Comment:
It is showing trailing white space error again but it is not showing in debugger.
How can I make sure that all the trailing white spaces are removed?